packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -243,10 +266,13 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
             return;
         }
 
-        const newText = await this.readContents();
-        if (newText === undefined || token.isCancellationRequested || this._dirty) {
+        let newText = await this.readContents();
```

Comment:
It's delivered as one compound event here first: https://github.com/eclipse-theia/theia/blob/b9987e8d121fb848a4e92257be3aaab053c7c24a/packages/filesystem/src/browser/file-resource.ts#L52-L54

It will be called once only if it manages to pass `FileResource.isInSync` check first: https://github.com/eclipse-theia/theia/blob/b9987e8d121fb848a4e92257be3aaab053c7c24a/packages/filesystem/src/browser/file-resource.ts#L115-L120

Reply:
you're saying one of those checks will prevent emitting of the event?